src/app/visualization-view/table-view/table-view.component.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * table-view.component.ts
+ * Component for displaying tabular summary of results.
+ *
+   Copyright 2018 Grzegorz Mrukwa
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
+import { ColumnDefinition } from '../column-definition';
+
+@Component({
+  selector: 'app-table-view',
+  templateUrl: './table-view.component.html',
+  styleUrls: ['./table-view.component.css']
+})
+export class TableViewComponent<T> implements OnInit, OnChanges {
+
+  @Input() columns: ColumnDefinition[];
+  @Input() data: T[];
+  private columnNames: string[] = [];
+
+  constructor() { }
+
+  ngOnInit() {
+    if (this.columns !== undefined) {
+      this.columnNames = this.columns.map(columnDefinition => columnDefinition.key);
+    }
+  }
+
+  ngOnChanges() {
+    if (this.columns !== undefined) {
+      this.columnNames = this.columns.map(columnDefinition => columnDefinition.key);
+    }
```

Comment:
Code from line 37,38 and 43,44 should be in one function, or maybe setter from this https://ngdev.space/angular-2-input-property-changes-detection-3ccbf7e366d2 ?
I do not check this solution, so I do not know if it works in our case.